### DISCUSSION TITLE
useRef in in preact-render-to-string

### DISCUSSION THREAD

**Discussion by fgnass** [preactjs/preact#3587] [Q&A] [answered]
I'm using @developit's prerender.js from `preact-iso` (a small wrapper around preact-render-to-string) in an SSG setup. 

I would like to use suspense-style data fetching in my components by throwing a promise. This works fine, and the components get re-rendered once the promise is resolved. What does not work is to use `useRef` to store a reference to the promise between render passes. I guess this is due to the fact that Preact resets the hook state?

The only workaround I found is to keep track of the promises outside the components, which feels super hacky. Can you think of a better way to implement this? Would it be possible to make useRef/useMemo work in RTS?

**Comment by JoviDeCroock** [comment, marked as answer]
This is the same in React though, hooks get reset between suspensions as it's the only way to safely handle WIP states, discarding them. What we did in [`urql`](https://github.com/FormidableLabs/urql/blob/main/packages/react-urql/src/hooks/cache.ts) was create an append-only cache to store the promise-->result transition.
